torch/_torch_docs.py

```diff
@@ -1721,6 +1721,11 @@ def merge_dicts(*dicts):
 tensor will be composed of lower-triangular Cholesky factors of each of the individual
 matrices.
 
+.. note:: ``torch.cholesky`` is deprecated. Please use :func:`torch.linalg.cholesky` instead.
```

Comment:
Let's not deprecate this, because torch.linalg.cholesky doesn't offer all the functionality that this does.

cc @IvanYashchuk 

Let's also update the linear algebra tracker to reflect that linalg.cholesky needs to be updated to have the functionality that cholesky does so we can deprecate cholesky. 

What we can say here is that users should prefer using the newer torch.linalg.cholesky when possible, but that it does not currently support the "upper" parameter.

Reply:
Do we want to support the "upper" parameter for `torch.linalg.cholesky`? NumPy doesn't have it (computes the lower triangular matrix always), SciPy has "lower" parameter (default `lower=False`, so upper-triangular matrix), in Julia language there is no such parameter (default is upper-triangular matrix).

Reply:
We need to support the functionality. It's unfortunate that NumPy and SciPy don't agree on which matrix to return, so I guess we'll be divergent from SciPy but have the same option it does to return lower or upper.